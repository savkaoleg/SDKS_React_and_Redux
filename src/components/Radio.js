import React from 'react'
import PropTypes from 'prop-types'
import './Radio.scss'

class Radio extends React.Component {

  onChange = () => {}

  onClick = (event) => {
    if (event.target.tagName !== 'INPUT'){
      this.props.handleRadioChange(this.props.value)
    }
  }

  render () {
    return (
      <div
        className="md-radio"
        onClick={this.onClick}
      >
        <input
          name={this.props.value}
          className={this.props.checked ? 'checked' : ''}
          type="radio"
          value={this.props.value}
          checked={this.props.checked}
          onChange={this.onChange}
        />
        <label>
          {this.props.children || this.props.value}
        </label>
      </div>
    )
  }
}

Radio.propTypes = {
  checked: PropTypes.bool,
  children: PropTypes.node,
  handleRadioChange: PropTypes.func,
  labelClass: PropTypes.string,
  value: PropTypes.string
}

export default Radio
