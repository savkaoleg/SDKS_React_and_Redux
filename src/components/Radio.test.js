import React from 'react'
import expect from 'expect'
import { shallow } from 'enzyme'
import Radio from './Radio'

describe('Radio', () => {
  it('Renders correctly', () => {
    const component = shallow(<Radio />)

    expect(component).toMatchSnapshot()
  })

  it('Value', () => {
    const testValue = 'abcde'
    const component = shallow(<Radio value={testValue} />)
    const result = component.find('input').prop('value')

    expect(result).toEqual(testValue)

  })

  it('Checked', () => {
    const testValue = false
    const component = shallow(<Radio checked={testValue} />)
    const result = component.find('input').prop('checked')

    expect(result).toEqual(testValue)

  })

  it('Checked className', () => {
    const testValue = true
    const component = shallow(<Radio checked={testValue} />)
    const result = component.find('input.checked').prop('checked')

    expect(result).toEqual(testValue)

  })

  it('Children', () => {
    const childrenText = 'some text'
    const children = (<div className='Children'>{childrenText}</div>)
    const component = shallow(<Radio value="some value">{children}</Radio>)
    const result = component.find('div.Children')

    expect(result.text()).toEqual(childrenText)
  })


  it('Spy', () => {
    const value = 'someValue'
    const component = shallow(<Radio value={value} />)

    const spy = spyOn(component.instance(), 'onClick')
    component.instance().forceUpdate()
    component.find('div.md-radio').simulate('click', {}, spy)
    expect(spy).toBeCalled()

  })
})
