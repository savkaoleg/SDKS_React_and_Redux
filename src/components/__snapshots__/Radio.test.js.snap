// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Renders correctly 1`] = `
<div
  className="md-radio"
  onClick={[Function]}
>
  <input
    className=""
    onChange={[Function]}
    type="radio"
  />
  <label />
</div>
`;
